core/datastorage/src/main/java/com/cognifide/aet/vs/mongodb/MongoDBClient.java

```diff
@@ -103,7 +103,15 @@ public void deactivate() {
 
     return companies;
   }
-  
+
+  protected void setupConfiguration(MongoDBClientConf config) {
+    mongoUri = Optional.of(config.mongoURI())
```

Comment:
I think `config.mongoURI()` can return null, don't we need `Optional.ofNullable` here?

Reply:
This is very odd for me, I had `ofNullable` at the beginning but my IDE prompted it as:
![2018-08-23_09h02_41](https://user-images.githubusercontent.com/19705838/44510053-cfc5be80-a6b3-11e8-91dc-da8b98421c6a.png)
I've tried to find any information if `@ObjectClassDefinition` and OSGi mechanics adds default by itself. Official Java docs: https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.6.2 won't mention explicitly what happens if you don't declare default value (I mean, that there will be `null` returned)...
When I checked it on debug, there actually was `null` - thanks for spotting that!
I will change that. I see that it looks like a good practice to return always some default value and I will set it to empty string in `mongoURI()`.